packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -173,6 +173,26 @@ export class ScopeHoistingPackager {
 
     res += this.outputFormat.buildBundlePostlude();
 
+    if (sourceMap) {
+      let promises = [];
+      // Traverse the bundle to get all source contents
+      this.bundle.traverseAssets(asset => {
```

Comment:
This doesn't seem quite right. Not all of the assets in the bundle will necessarily be included in the final output. Some might be skipped e.g. due to no used symbols. Also, shouldn't the sourcemap library handle this when merging the source maps from the original asset?

Reply:
I think you're right the new scope hoisting no longer needs this as now you're using the sourcemap library already and not relying on the AST for all of this.

Previously we were relying on the Babel AST for this and the AST does not keep a reference to the source content so we had to extract that from the maps and manually add this. We can probably refactor this a bit further to make it a bit more lightweight now that the scope-hoisting implementation already supports the source-maps properly.